### DISCUSSION TITLE
Is it possible to convert a dgn file into bim file without using iTwinSnapshot app?

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#3652] [Q&A]
I want to write a task that automaticly and continully coverting dgn files into bim files so that I could gather the features of imodel to update my predict model. Is this possible?

**Comment by pmconne** [comment]
Yes. iTwinSnapshot is run once against a set of input files to produce a static ("snapshot") iModel of the state of those files at that time. Under the hood it processes the input files using one or more [connectors](https://www.itwinjs.org/learning/imodel-connectors/).

Continually reconverting the same input files to produce new iModels is an inefficient process and retains no record of the changes to the input files between runs.

A better way is to continuously [synchronize](https://communities.bentley.com/products/digital-twin-cloud-services/itwin-services/w/synchronization-wiki/47596/ways-to-sync-your-data-to-an-itwin) your input files with the **same** iModel. Each time you synchronize, the connectors detect what has changed in the input files and updates the contents of your iModel appropriately, retaining a record of those changes. This is also faster than reconverting everything from scratch every time.

**Reply by ayushibasaiwal** [reply]
@Snailya If your query is resolved, can you please mark this reply as an answer, or in case you need any help, please ask.
